service/controller/resource/namespaceauth/delete.go

```diff
@@ -0,0 +1,56 @@
+package namespaceauth
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/rbac-operator/service/controller/key"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	namespace, err := key.ToNamespace(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = r.k8sClient.RbacV1().RoleBindings(namespace.Name).Get(viewAllRole, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		// do nothing
+	} else {
+		if err != nil {
+			return microerror.Mask(err)
+		} else {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "deleting view role binding")
+
+			err = r.k8sClient.RbacV1().RoleBindings(namespace.Name).Delete(viewAllRole, &metav1.DeleteOptions{})
+			if err != nil {
```

Comment:
Maybe catch `IsNotFound` here just in case? Shouldn't happen nearly ever, but similar to `AlreadyExists` in creation. Same thing below for `Role`.

Reply:
Good idea, thx!